cetloxblock/static/js/services.js

```diff
@@ -0,0 +1,202 @@
+﻿/* data queries from CET for CetLoXBlock. */
+/// <reference path="http://ajax.googleapis.com/ajax/libs/jquery/2.2.4/jquery.js" />
+/// <reference path="http://cdn.cet.ac.il/ui-services/login/js/cet.ui-services.login.full.js" />
+function CetServices() {
+
+  this.getCetDomain = function () {
+    domainsMap = {
+      "dev": ".dev.cet.ac.il",
+      "testing": ".testing.cet.ac.il",
+      "prod": ".cet.ac.il"
+    };
+    function getEnv() {
+      var host = window.location.host.toLowerCase();
+      if (host.indexOf("localhost") >= 0 || host.indexOf("127.0.0.1") >= 0) {
+        return "dev";
+      }
+      if (host.indexOf('campus-dev.') >= 0) {
+        return "testing";
+      }
+      return "prod";
+    }
+    return domainsMap[getEnv()];
+  }
+  var CetDomain = this.CetDomain = this.getCetDomain();
+
+  var PRODUCT_PLAYER = "//productplayer" + CetDomain + "/";
+  var PRODUCT_PLAYER_API = "//productplayer" + CetDomain + "/api/ApiProxy/";
+  var PRODUCT_PLAYER_SERVICES = "//productplayerservices" + CetDomain + "/";
+  var TREE_SERVICE = "//treeservice" + CetDomain + "/api/Trees/";
+  var TREE_LMS_SERVICE = "//treeservice" + CetDomain + "/api/TreeLMS/";
+  var DEFAULT_CET_COURSE = "Edx-adaptive-project";
+
+  this.getCourseEdxName = function () {
+    var match = /-v\d+:([^\/\+]+\+[^\/\+]+\+[^\/\+]+)[\/\+]/gi.exec(location.pathname);
+    if (match.length) {
+      return match[1];
+    }
+  }

```

Comment:
I'm pretty sure you can get the current course ID from the DOM, [like we do here](https://github.com/open-craft/problem-builder/blob/fc465870ef67b152fe4f60d7b9865f8748911a90/problem_builder/public/js/instructor_tool.js#L263).

Reply:
This was very helpful. I expected data of this type to be documented somewhere (which unfortunately I did not find), so I know I can count on it. Nevertheless the url format is hardly safer.

Reply:
Your suggestion is working in the campus, but fails on devstack (the MASTER that I installed when I started the project). Feeling it a bit shaky, I left the previous algorithm as a fallback.

Reply:
FYI, if you develop using the [Campus devstack](https://gitlab.com/opencraft/client/campus/campus-devstack) (which runs [`opencraft-release/ginkgo.1-campus`|https://github.com/open-craft/edx-platform/tree/opencraft-release/ginkgo.1-campus], then you'll find it easier to debug issues on deployment.

Also note that edX have [removed the vagrant devstack in favour of Docker](https://github.com/edx/configuration/pull/4492), so if you have to rebuild your `master` stack, you may run into issues.